mmv1/products/discoveryengine/Schema.yaml

```diff
@@ -0,0 +1,101 @@
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.dataStores.schemas'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas?schemaId={{schema_id}}'
+update_verb: :PATCH
+update_mask: true
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}'
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}',
+  ]
+
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_schema_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_schema%s", context["random_suffix"])'
+    vars:
+      data_store_id: "data-store-id"
+      schema_id: "schema-id"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The geographic location where the data store should reside. The value can
+      only be one of "global", "us" and "eu".
+  - !ruby/object:Api::Type::String
+    name: 'dataStoreId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The unique id of the data store.
+  - !ruby/object:Api::Type::String
+    name: 'schemaId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The unique id of the schema.
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the schema. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/dataStores/{data_store_id}/schemas/{schema_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'jsonSchema'
+    description: |
+      The JSON representation of the schema.
```

Comment:
`exactly_one_of` should be used for the property `jsonSchema`, even `structSchema` is not added to the yaml file.

Reply:
Thanks, added. I have a question about how to use `exactly_one_of` correctly. 

Option 1: list only `json_schema` because we do not have field `structSchema`
```
name: 'jsonSchema'
exactly_one_of:
  - json_schema
```
Option 2 [what I chose]: list both `json_schema` and `struct_schema` even we do not have field `structSchema`
```
name: 'jsonSchema'
exactly_one_of:
  - struct_schema
  - json_schema
```

I tried to test these two options locally and found no difference.

Reply:
The difference should be on [this line](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11018-old..auto-pr-11018#diff-69a1a182e5b6ca68d441a7d74ce34b6feb9328997c430a7659cd24ba3851c1cdR85) of the generated provider code:
```
ExactlyOneOf: []string{"json_schema"},
```
vs

```
ExactlyOneOf: []string{"json_schema", "struct_schema"},
```

Option 1 should be correct, as `struct_schema` has not been supported yet.

This is the [doc](https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#exactly_one_of) for `exactly_one_of`

Reply:
Once `struct_schema` is supported later, the option 2 should be used and `exactly_one_of` should be added to both fields `jsonSchema` and `structSchema `

Reply:
Thanks, I see. Removed `struct_schema` here. 

Given https://yaqs.corp.google.com/eng/q/1294165867822579712, we plan not to support `struct_schema` because we cannot guarantee the max nesting depth of this recursive pattern `google.protobuf.Struct`.